sklearn/model_selection/_search.py

```diff
@@ -320,24 +324,61 @@ def fit_grid_point(X, y, estimator, parameters, train, test, scorer,
     return score, parameters, n_samples_test
 
 
+def _check_param_dist(param_dist):
```

Comment:
> Indeed I missed the use case of any arbitrary iterable!
 
It's not a public API use-case, nor tested, so that's fair enough. But the code hints at its current agnosticism towards such iterables.